docs/workflow.md

```diff
@@ -0,0 +1,149 @@
+<!--
+TODO: should we call "this project" Juno? I saw in communication.md we did,
+but I'm not sure
+ -->
+
+# Workflow
+
+One of the goals of this project is to help users adopt a more efficient
+workflow when editing Julia code. To understand how to gain this productivity
+boost, we will consider a common workflow for many Julia developers and a Juno
+based workflow.
+
+## Status quo
```

Comment:
While I kinda like the comparison to a plain-julia-workflow, I'm not entirely sure describing the status quo is necessary. It definitely is a nice read and now everyone knows why a Juno-based workflow is just plain superior ;)